addons/mail/static/src/components/composer/composer.scss

```diff
@@ -13,258 +14,76 @@
 
     &.o-has-current-partner-avatar {
         grid-template-columns: 50px 1fr;
-        padding: map-get($spacers, 3) map-get($spacers, 3) map-get($spacers, 4) map-get($spacers, 1);
-
-        &:not(.o-has-footer) {
-            padding-bottom: map-get($spacers, 4);
-        }
-
-        &:not(.o-has-header) {
-            padding-top: map-get($spacers, 4);
-
-            &.o-is-in-thread-view {
-                padding-top: map-get($spacers, 1);
-                padding-bottom: map-get($spacers, 1);
-            }
-        }
-    }
-}
-
-.o_Composer_actionButtons {
-    &.o-composer-is-compact {
-        display: flex;
-    }
-    &:not(.o-composer-is-compact) {
-        margin-top: map-get($spacers, 2);
     }
 }
 
 .o_Composer_attachmentList {
-    flex: 1 1 auto;
-
     &.o-composer-is-compact {
         max-height: 100px;
     }
 
     &:not(.o-composer-is-compact) {
-        overflow-y: auto;
         max-height: 300px;
     }
 }
 
 .o_Composer_buttons {
-    display: flex;
-    align-items: stretch;
-    align-self: stretch;
-    flex: 0 0 auto;
-    min-height: 41px; // match minimal-height of input, including border width
-
-    &:not(.o-composer-is-compact) {
-        border: 0;
-        height: auto;
-        padding: 0 10px;
-        width: 100%;
-    }
+    min-height: $o-mail-composer-text-input-height + $border-width; // match minimal-height of input, including border width
 }
 
 .o_Composer_coreFooter {
     grid-area: core-footer;
```

Comment:
~~seems like each element assigned to a grid area always uses the area that follows its name.
`coreFooter -> core-footer`, `coreHeader -> core-header`... what about a loop?~~

~~something like~~
```scss
$-areas: Header, Footer, Main ... ;

@each  $area-name in $areas {
    .o_Composer_core{$area-name} {
        grid-area: core-{$area-name} ;
    }
}
```
---

nevermind , it's 9 lines against 6... not worth since except `.o_Composer_coreHeader {}`  you'll have to create these selectors anyway to apply other rules.